minecraft_script/compiler/compile_interpreter.py

```
@@ -58,12 +58,25 @@ def __repr__(self) -> str:
         return f'CompileSymbols({self.parent !r})'
 
 class CompileContext:
-    def __init__(self, mcfunction_name: str = None, *, parent: "CompileContext" = None, top_level: bool = False):
+    def __init__(
+        self,
+        mcfunction_name: str = None,
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)